app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsFragment.kt

```diff
@@ -509,18 +508,30 @@ class PaymentMethodsFragment : DaggerFragment(), PaymentMethodsView {
     bonus_layout.visibility = View.VISIBLE
     bonus_msg.visibility = View.VISIBLE
     no_bonus_msg?.visibility = View.INVISIBLE
+    bottom_separator?.visibility = View.VISIBLE
+  }
+
+  override fun removeBonus() {
+    bonus_layout.visibility = View.GONE
+    bonus_msg.visibility = View.GONE
+    no_bonus_msg?.visibility = View.GONE
+    bottom_separator?.visibility = View.GONE
   }
 
   override fun hideBonus() {
-    bonus_layout.visibility = View.INVISIBLE
-    bonus_msg.visibility = View.INVISIBLE
-    bottom_separator?.visibility = View.INVISIBLE
+    if (bonus_layout.visibility != View.GONE) {
```

Comment:
Control should not be done by the value of the visibility but by the conditions of the calling of the method